core/src/main/java/hudson/Functions.java

```diff
@@ -1306,7 +1309,7 @@ public static ThreadGroupMap sortThreadsAndGetGroupMap(ThreadInfo[] list) {
     private static class ThreadSorterBase {
         protected Map<Long,String> map = new HashMap<>();
 
-        private ThreadSorterBase() {
+        public ThreadSorterBase() {
```

Comment:
out of curiosity: why did you change it?

Reply:
it needed a public constructor to be serializable.